compiler/rustc_codegen_llvm/src/llvm/enzyme_ffi.rs

```diff
@@ -0,0 +1,39 @@
+#![allow(non_camel_case_types)]
+
+use libc::{c_char, c_uint, size_t};
+
+use super::ffi::{Attribute, BasicBlock, Builder, Metadata, Module, Type, Value};
+extern "C" {
+    // Enzyme
+    pub fn LLVMRustHasMetadata(I: &Value, KindID: c_uint) -> bool;
+    pub fn LLVMRustEraseInstBefore(BB: &BasicBlock, I: &Value);
+    pub fn LLVMRustGetLastInstruction<'a>(BB: &BasicBlock) -> Option<&'a Value>;
+    pub fn LLVMRustDIGetInstMetadata(I: &Value) -> &Metadata;
+    pub fn LLVMRustEraseInstFromParent(V: &Value);
+    pub fn LLVMRustGetTerminator<'a>(B: &BasicBlock) -> &'a Value;
+    pub fn LLVMGetFunctionCallConv(F: &Value) -> c_uint;
+
+    pub fn LLVMGetReturnType(T: &Type) -> &Type;
+    pub fn LLVMDumpModule(M: &Module);
+    pub fn LLVMCountStructElementTypes(T: &Type) -> c_uint;
+    pub fn LLVMVerifyFunction(V: &Value, action: LLVMVerifierFailureAction) -> bool;
```

Comment:
This is `LLVMBool` not `bool`, so should use `Bool` from `ffi.rs` or will eventually crash.

Reply:
Plus, i'm not sure that c and cpp enums are the same. Probably should be added via wrapper like other enums in RustWrapper.cpp